tests/rptest/tests/upgrade_test.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2022 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import re
+
+from ducktape.utils.util import wait_until
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.cluster import cluster
+from rptest.services.redpanda import RESTART_LOG_ALLOW_LIST
+from rptest.services.redpanda_installer import RedpandaInstaller
+
+
+def wait_for_num_versions(redpanda, num_versions):
+    def get_unique_versions():
+        node = redpanda.nodes[0]
+        brokers_list = \
+            str(node.account.ssh_output(f"{redpanda.find_binary('rpk')} redpanda admin brokers list"))
+        redpanda.logger.debug(brokers_list)
+        version_re = re.compile("v\\d+\\.\\d+\\.\\d+")
+        return set(version_re.findall(brokers_list))
+
+    # NOTE: allow retries, as the version may not be available immediately
+    # following a restart.
+    wait_until(lambda: len(get_unique_versions()) == num_versions,
+               timeout_sec=30)
+    unique_versions = get_unique_versions()
+    assert len(unique_versions) == num_versions, unique_versions
+    return unique_versions
+
+
+class UpgradeFromSpecificVersion(RedpandaTest):
+    """
+    Basic test that upgrading software works as expected.
+    """
+    def __init__(self, test_context):
+        super(UpgradeFromSpecificVersion,
+              self).__init__(test_context=test_context, num_brokers=3)
+        self.installer = RedpandaInstaller(self.redpanda, self.test_context)
```

Comment:
Since RedpandaInstaller requires a RedpandaService TestContext instance, consider creating the installer automatically within RedpandaService.

Reply:
Sure, will do